includes/plugins/wc-memberships/class-memberships.php

```diff
@@ -774,15 +773,18 @@ public static function render_js() {
 				ras.on( 'reader', function( ev ) {
 					if ( ev.detail.authenticated && ! window?.newspackReaderActivation?.getPendingCheckout() ) {
 						if ( ras.overlays.get().length ) {
+							// When an overlay is added or removed,
+							// check if there are none – this means an overlay was removed and there are none left.
+							// In this case, reload the window.
 							ras.on( 'overlay', function( ev ) {
-								if ( ! ev.detail.overlays.length && ! window?.newspackReaderActivation?.openNewslettersSignupModal ) {
```

Comment:
@chickenn00dle – please advise if this condition was needed. It appears to be a constant condition (openNewslettersSignupModal is always defined), so the reload would never happen. I can't see the reason for this condition, and there's no comment to shed light on it.

Reply:
Hey @adekbadek,

This value will be set when the post checkout newsletters modal is active. Unfortunately when transitioning from checkout modal to newsletters modal, there is a moment when overlays are empty, so if we don't add this check, the page will reload between these steps for relevant product.

Reply:
> This value will be set when the post checkout/registration newsletters modal is active.

In my testing, this value is always set, so it's a constant condition. It's [included here](https://github.com/Automattic/newspack-plugin/blob/trunk/src/reader-activation/index.js#L416-L440) where `readerActivation` is defined, and then that value is [assigned to the global](https://github.com/Automattic/newspack-plugin/blob/trunk/src/reader-activation/index.js#L484). 

> when transitioning from registration modal to newsletters modal, there is a moment when overlays are empty

Got it, makes sense. I've updated the code to accommodate this condition in c4785979268c